pages/en/lb4/Decorator.md

```diff
@@ -0,0 +1,125 @@
+---
+lang: en
+title: 'Decorators'
+keywords: LoopBack 4.0, LoopBack-Next
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Decorators.html
+summary:
+---
+
+A decorator allows you annotate or modify your class declarations and members with metadata.
+
+## Introduction
+
+*To get familiar with this TypeScript feature, try read [Understand TypeScript Decorator](#understand-typescript-decorator)*
+
+Decorator is a good strategy to keep your code loosely coupled and modifies metadata at runtime. Currently LoopBack core provides built-in decorator functions you can use directly by `@myDecorator`, e.g. Route decorators that introduces in section Route-Decorators, and implements a dependency injection system that can apply a decorator by `@inject('myDecorator')`. 
+
+Some components also exports decorators for you:
+- [authentication](#authentication-decorators)
+- [repository](#repository-decorators)
+
+*Janny: Most examples are documented in a certain concept that a decorator works closely with, I am not sure should I turn this page to something like a reference:*
+
+- some usage are well documented, I linked to certain section
+- some usage can be found in unit test, while integration example is missing, should I create on this page?
+- some have integration test, modify it and add on this page?
+- Do I need to document what metadata each decorator adds to the reflector and document in the reference?
+- I am confused with some decorator can be used by either @mydecorator or @inject(‘mydecorator’), are they equal? 
+
+## Route Decorators
+- @get, @post, @put, @patch, @del
+
+  Register route for a controller by rest decorator, for their usage, refer to [Routing to Controllers](controller.htm#Routing-to-Controllers)
+
+- @api
+
+  Register route for a controller by api decorator, more details and example refer to [Specifying Controller APIs](controllers.htm#Specifying Controller APIs)
+
+- @operation
+
+  Expose a Controller method as a REST API operation. Usage refer to [code](https://github.com/strongloop/loopback-next/blob/debe221ee7eb07b6d6962416b79592c8f13d1e91/packages/core/src/router/metadata.ts#L230-L234)
+
+  *Janny: Need example in doc?*
```

Comment:
Nope, it's documented in several areas. But I wouldn't use the link you have here, as it's tied to a specific commit. `/blob/debe221ee7eb07b6d6962416b79592c8f13d1e91/packages...` should be `/blob/master/packages...`

Reply:
Additionally, we're linking to a document called metadata. If there isn't anything that describes how to do this in the Controller section, then that's a problem to fix later.

Reply:
Currently the examples in concept [controllers](http://loopback.io/doc/en/lb4/Controllers.html#routing-to-controllers) does contain `@operation`, so I will create another PR to add it to controllers section.